### PR TITLE
Transfer nft api fix

### PR DESCRIPTION
**Pull request by rex-god**
- update nft api fix
- transfer nft and nft_detach api fixed.
- some code refactoring

### CODE REVIEW

File: src/modules/nft/components/transfer-nft/transfer-nft.service.ts [hunk 1]
```diff
@@ -46,56 +66,35 @@ export class TransferNftService {
       const toAccount = await getOrCreateAssociatedTokenAccount(
         connection,
         fromKeypair,
-        tokenAddress,
+        new PublicKey(tokenAddress),
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/transfer-nft.service.ts:69]: tokenAddress is already of type publicKey, as per line no. 39.

File: src/modules/nft/components/transfer-nft/transfer-nft.service.ts [hunk 2]
```diff
@@ -146,48 +156,72 @@ export class TransferNftService {
...
 
-      const serializedTransaction = tx.serialize({ requireAllSignatures: false, verifySignatures: false });
+      const serializedTransaction = tx.serialize({
+        requireAllSignatures: false,
+        verifySignatures: false,
+      });
       const transactionBase64 = serializedTransaction.toString('base64');
       return { encoded_transaction: transactionBase64 };
     } catch (error) {
       console.log(error);
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/transfer-nft.service.ts:171]: We can remove error log from here.

File: src/modules/nft/components/transfer-nft/transfer-nft.service.ts [hunk 3]
```diff
@@ -146,48 +156,72 @@ export class TransferNftService {
...
+  }
+
+  hasUpdateAuthority(
+    fromAddress: PublicKey,
+    currentUpdateAuthority: PublicKey,
+  ): boolean {
+    return fromAddress.toBase58() === currentUpdateAuthority.toBase58();
+  }
+
+  async getUpdateAuthorityChangeInstruction(
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/transfer-nft.service.ts:183]: should we move this function to Utils?

File: src/modules/nft/components/transfer-nft/transfer-nft.service.ts [hunk 4]
```diff
@@ -46,56 +62,43 @@ export class TransferNftService {
...
-          royalty: nft.sellerFeeBasisPoints,
-          symbol: nft.symbol,
-          token_address: nft.mint.toBase58(),
-          update_authority: nft.updateAuthority.toBase58(),
-          new_update_authority: toAddress.toBase58(),
-          private_key: fromAdress,
-        });
-      }
-      return { updateTxId: updateTxId?.txId ?? 'update_authority not transfered', transferTxId: transferTxId };
+      await connection.confirmTransaction(transferTxId).catch((r) => {
```
Comment (codinpsycho) [inline src/modules/nft/components/transfer-nft/transfer-nft.service.ts:89]: Why not put this in a try-catch block? Instead of a .catch

**Review by codinpsycho — APPROVED** [review]

**Pull request merged**